### ISSUE TITLE
backendutil: Match evaluates LARGER using >= instead of >

### ISSUE THREAD

**Issue by drauschenbach** [emersion/go-imap#298]
The code:

```go
if c.Larger > 0 && uint32(n) < c.Larger {
	return false, nil
}
```

should read:

```go
if c.Larger > 0 && uint32(n) <= c.Larger {
	return false, nil
}
```

**Comment by drauschenbach** [comment]
Oh, and same for SMALLER

**Comment by emersion** [comment]
Hmm, I can't find where this is specified in the RFC… Do you have a good reference?

**Comment by drauschenbach** [comment]
https://tools.ietf.org/html/rfc3501#section-6.4.4

```
LARGER <n>
         Messages with an [RFC-2822] size larger than the specified
         number of octets.
```

**Comment by emersion** [comment]
Yeah, but this doesn't specify whether it's "strictly larger" or "larger or equal".

Seems like Dovecot does a strict check (https://github.com/dovecot/core/blob/91d9e32ea0831f4e0b9bf158453ec0559659bb48/src/lib-storage/index/index-search.c#L389), so I suppose it's the right thing to do. Could you send a PR?

**Comment by drauschenbach** [comment]
Interesting. I see no possible ambiguity in the English language. But you're Finnish, right?

UW-IMAP is the reference implementation created to clarify such ambiguities.

Looking at http://ftp.ntua.gr/pub/net/mail/imap/imap-2007f.tar.gz src/c-client/mail.c:3473-3474 yields the following which supports my assumptions:

```c
/* size ranges */
  if ((pgm->larger && (elt->rfc822_size <= pgm->larger)) ||
```

**Comment by drauschenbach** [comment]
Sure thing, I'll prepare a PR.

**Comment by emersion** [comment]
>I see no possible ambiguity in the English language. But you're Finnish, right?

I'm French, and maybe that's the reason why I feel an ambiguity indeed! We routinely use "lower than" to refer to <= (and "strictly lower than" for <).

Anyway, that's a good catch, thanks!

**Comment by Neopallium** [comment]
Needs to also include the message headers.  Right now it is only using the body size.  Dovecot uses the whole message (body and headers) for the size.  I also fixed this in PR #300 in commit 3edd151759405de2e4fb4b2b57fa34eedff90411.
